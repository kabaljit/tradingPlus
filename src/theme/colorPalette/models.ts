import {  nestedColorPalette, } from './index';

export type ColorPalette = typeof nestedColorPalette;

export type ColorTones = ColorPalette[keyof ColorPalette];

export type ColorValue = ColorTones[0];