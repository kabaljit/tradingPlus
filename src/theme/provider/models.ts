import { ColorPalette } from '../palette.models';
import { IColorContexts } from '../context.models';

export interface ITheme {
  colorPalette: ColorPalette;
  context: {
    colors: IColorContexts;
  };
}

export interface IThemeProvider {
  theme?: ITheme;
}
