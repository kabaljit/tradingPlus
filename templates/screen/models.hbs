export interface {{ pascalCase name }}ScreenParams {
  // Navigation parameters here.
}

export interface {{ pascalCase name }}ScreenFormValues {
  // Formik parameters here.
}

export interface {{ pascalCase name }}ScreenProps extends {{ pascalCase name }}ScreenParams {
}
