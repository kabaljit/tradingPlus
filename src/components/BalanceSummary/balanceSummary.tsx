import * as React from "react";

import { BalanceSummaryProps } from "./balanceSummary.models";
import { BalanceSummaryView } from "./balanceSummary.styles";
import { i18n } from "./balanceSummary.i18n";
import { P } from "../Typography";
import { Box, Row } from "../Box";

export const BalanceSummary: React.FunctionComponent<BalanceSummaryProps> = ({}) => {
  return (
    <BalanceSummaryView>
      <Box spacing={{ top: 4, right: 4, left: 4, bottom: 4 }}>
        <Row flexDirection="row" spacing={{ bottom: 0 }}>
          <P weight="bold">{i18n.t("totalLabel")}: </P>
          <P>£0</P>
        </Row>
        <Row flexDirection="row">
          <P weight="bold">{i18n.t("portfolioLabel")}: </P>
          <P>£0</P>
        </Row>
      </Box>
    </BalanceSummaryView>
  );
};
