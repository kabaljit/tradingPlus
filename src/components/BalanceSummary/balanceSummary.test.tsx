import React from 'react';
import renderer from 'react-test-renderer';

import { BalanceSummary } from './balanceSummary';
import { BalanceSummaryProps } from './balanceSummary.models';

const props: BalanceSummaryProps = {

};

describe('Given a BalanceSummary component, when it is rendered', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<BalanceSummary {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
