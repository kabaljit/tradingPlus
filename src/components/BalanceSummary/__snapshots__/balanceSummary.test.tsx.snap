// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a BalanceSummary component, when it is rendered should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#5DA6EC",
        "borderRadius": 13,
        "width": "100%",
      },
    ]
  }
>
  <View
    spacing={
      Object {
        "bottom": 4,
        "left": 4,
        "right": 4,
        "top": 4,
      }
    }
    style={
      Array [
        Object {
          "paddingBottom": 21.25714285714286,
          "paddingLeft": 21.25714285714286,
          "paddingRight": 21.25714285714286,
          "paddingTop": 21.25714285714286,
        },
      ]
    }
    theme={
      Object {
        "colorPalette": Object {
          "green": Array [
            "#0BA866",
          ],
          "grey": Array [
            "#F7F8FC",
            "#EDEDF4",
            "#E1E1EA",
            "#D4D5DD",
            "#BCBCC9",
            "#9699AA",
            "#7C7E8D",
            "#676978",
            "#484A58",
          ],
          "mango": Array [
            "#FFB900",
            "#FFD566",
          ],
          "midnightBlue": Array [
            "#262835",
          ],
          "red": Array [
            "#FFFEF3",
            "#FFE0D7",
            "#FFC2BB",
            "#FFA5A0",
            "#FF8886",
            "#FF6B6C",
            "#FF4D54",
            "#FF333C",
            "#FF1A23 ",
          ],
          "seaGreen": Array [
            "#D5F1F3",
            "#C1EAED",
            "#ACE3E7",
            "#9CDEE1",
            "#82D5DA",
            "#6DCDD3",
            "#59C7CE",
            "#2FB9C2",
            "#238B92",
            "#185D61",
          ],
          "skyBlue": Array [
            "#F7FAFD",
            "#E5F2FC",
            "#C0DEF8",
            "#99CAF4",
            "#75B6EF",
            "#5DA6EC",
            "#4C97E8",
            "#4689DA",
            "#3871C2",
            "#3260B0",
          ],
          "white": Array [
            "#FFFFFF",
          ],
          "yellow": Array [
            "#FFF8E4",
          ],
        },
        "context": Object {
          "colors": Object {
            "actions": Object {
              "background": Object {
                "link": Object {
                  "complete": "",
                  "default": "transparent",
                  "disabled": "transparent",
                  "error": "",
                  "focused": "transparent",
                  "loading": "transparent",
                },
                "primary": Object {
                  "complete": "",
                  "default": "#FF4D54",
                  "disabled": "#E1E1EA",
                  "error": "",
                  "focused": "",
                  "loading": "#E1E1EA",
                },
                "secondary": Object {
                  "complete": "",
                  "default": "transparent",
                  "disabled": "transparent",
                  "error": "",
                  "focused": "transparent",
                  "loading": "transparent",
                },
              },
              "foreground": Object {
                "link": Object {
                  "complete": "",
                  "default": "#FFFEF3",
                  "disabled": "#262835",
                  "error": "",
                  "focused": "",
                },
                "primary": Object {
                  "complete": "",
                  "default": "#FFFFFF",
                  "disabled": "#262835",
                  "error": "",
                  "focused": "",
                },
                "secondary": Object {
                  "complete": "",
                  "default": "#FFFEF3",
                  "disabled": "#262835",
                  "error": "",
                  "focused": "",
                },
              },
            },
            "input": Object {
              "background": Object {
                "complete": "#F7FAFD",
                "default": "#F7F8FC",
                "error": "#FFFEF3",
                "focused": "#FFFFFF",
              },
              "border": Object {
                "complete": "#4689DA",
                "default": "#9699AA",
                "error": "#FFE0D7",
                "focused": "#FF4D54",
              },
              "icon": Object {
                "complete": "#262835",
                "default": "#262835",
                "error": "#262835",
                "focused": "#262835",
              },
              "label": Object {
                "complete": "#9699AA",
                "default": "#9699AA",
                "error": "#FFE0D7",
                "focused": "#9699AA",
              },
              "outerLabel": Object {
                "complete": "#262835",
                "default": "#262835",
                "error": "#262835",
                "focused": "#262835",
              },
            },
            "states": Object {
              "error": Object {
                "complete": "",
                "default": "#FFFEF3",
                "error": "",
                "focused": "",
              },
              "loading": Object {
                "complete": "",
                "default": "#FFB900",
                "error": "",
                "focused": "",
              },
            },
            "typography": Object {
              "blue": "#3260B0",
              "error": "#FF8886",
              "inverse": "#D4D5DD",
              "link": "#3871C2",
              "mango": "#FFB900",
              "primary": "#262835",
              "secondary": "#F7F8FC",
              "success": "#0BA866",
              "tertiary": "#7C7E8D",
              "white": "#FFFFFF",
            },
          },
          "fontSizes": Object {
            "paragraph": Object {
              "large": 31.5,
              "normal": 27.5,
              "small": 23.5,
            },
            "title": Object {
              "large": 63,
              "normal": 47.5,
              "small": 35.5,
            },
          },
          "fontWeights": Object {
            "paragraph": Object {
              "bold": "700",
              "medium": "600",
              "regular": "400",
            },
            "title": Object {
              "bold": "700",
              "medium": "600",
              "regular": "400",
            },
          },
          "sizes": Object {
            "buttonHeight": 108.5,
            "formElementHeight": 91,
            "formElementIconHeight": 67,
            "hairline": 0.5,
            "listItemHeight": 108.5,
          },
          "spacing": Object {
            "element": 16,
            "screenEdge": 31.5,
          },
        },
      }
    }
  >
    <View
      flexDirection="row"
      spacing={
        Object {
          "bottom": 0,
        }
      }
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingBottom": 2.6571428571428575,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
        ]
      }
      theme={
        Object {
          "colorPalette": Object {
            "green": Array [
              "#0BA866",
            ],
            "grey": Array [
              "#F7F8FC",
              "#EDEDF4",
              "#E1E1EA",
              "#D4D5DD",
              "#BCBCC9",
              "#9699AA",
              "#7C7E8D",
              "#676978",
              "#484A58",
            ],
            "mango": Array [
              "#FFB900",
              "#FFD566",
            ],
            "midnightBlue": Array [
              "#262835",
            ],
            "red": Array [
              "#FFFEF3",
              "#FFE0D7",
              "#FFC2BB",
              "#FFA5A0",
              "#FF8886",
              "#FF6B6C",
              "#FF4D54",
              "#FF333C",
              "#FF1A23 ",
            ],
            "seaGreen": Array [
              "#D5F1F3",
              "#C1EAED",
              "#ACE3E7",
              "#9CDEE1",
              "#82D5DA",
              "#6DCDD3",
              "#59C7CE",
              "#2FB9C2",
              "#238B92",
              "#185D61",
            ],
            "skyBlue": Array [
              "#F7FAFD",
              "#E5F2FC",
              "#C0DEF8",
              "#99CAF4",
              "#75B6EF",
              "#5DA6EC",
              "#4C97E8",
              "#4689DA",
              "#3871C2",
              "#3260B0",
            ],
            "white": Array [
              "#FFFFFF",
            ],
            "yellow": Array [
              "#FFF8E4",
            ],
          },
          "context": Object {
            "colors": Object {
              "actions": Object {
                "background": Object {
                  "link": Object {
                    "complete": "",
                    "default": "transparent",
                    "disabled": "transparent",
                    "error": "",
                    "focused": "transparent",
                    "loading": "transparent",
                  },
                  "primary": Object {
                    "complete": "",
                    "default": "#FF4D54",
                    "disabled": "#E1E1EA",
                    "error": "",
                    "focused": "",
                    "loading": "#E1E1EA",
                  },
                  "secondary": Object {
                    "complete": "",
                    "default": "transparent",
                    "disabled": "transparent",
                    "error": "",
                    "focused": "transparent",
                    "loading": "transparent",
                  },
                },
                "foreground": Object {
                  "link": Object {
                    "complete": "",
                    "default": "#FFFEF3",
                    "disabled": "#262835",
                    "error": "",
                    "focused": "",
                  },
                  "primary": Object {
                    "complete": "",
                    "default": "#FFFFFF",
                    "disabled": "#262835",
                    "error": "",
                    "focused": "",
                  },
                  "secondary": Object {
                    "complete": "",
                    "default": "#FFFEF3",
                    "disabled": "#262835",
                    "error": "",
                    "focused": "",
                  },
                },
              },
              "input": Object {
                "background": Object {
                  "complete": "#F7FAFD",
                  "default": "#F7F8FC",
                  "error": "#FFFEF3",
                  "focused": "#FFFFFF",
                },
                "border": Object {
                  "complete": "#4689DA",
                  "default": "#9699AA",
                  "error": "#FFE0D7",
                  "focused": "#FF4D54",
                },
                "icon": Object {
                  "complete": "#262835",
                  "default": "#262835",
                  "error": "#262835",
                  "focused": "#262835",
                },
                "label": Object {
                  "complete": "#9699AA",
                  "default": "#9699AA",
                  "error": "#FFE0D7",
                  "focused": "#9699AA",
                },
                "outerLabel": Object {
                  "complete": "#262835",
                  "default": "#262835",
                  "error": "#262835",
                  "focused": "#262835",
                },
              },
              "states": Object {
                "error": Object {
                  "complete": "",
                  "default": "#FFFEF3",
                  "error": "",
                  "focused": "",
                },
                "loading": Object {
                  "complete": "",
                  "default": "#FFB900",
                  "error": "",
                  "focused": "",
                },
              },
              "typography": Object {
                "blue": "#3260B0",
                "error": "#FF8886",
                "inverse": "#D4D5DD",
                "link": "#3871C2",
                "mango": "#FFB900",
                "primary": "#262835",
                "secondary": "#F7F8FC",
                "success": "#0BA866",
                "tertiary": "#7C7E8D",
                "white": "#FFFFFF",
              },
            },
            "fontSizes": Object {
              "paragraph": Object {
                "large": 31.5,
                "normal": 27.5,
                "small": 23.5,
              },
              "title": Object {
                "large": 63,
                "normal": 47.5,
                "small": 35.5,
              },
            },
            "fontWeights": Object {
              "paragraph": Object {
                "bold": "700",
                "medium": "600",
                "regular": "400",
              },
              "title": Object {
                "bold": "700",
                "medium": "600",
                "regular": "400",
              },
            },
            "sizes": Object {
              "buttonHeight": 108.5,
              "formElementHeight": 91,
              "formElementIconHeight": 67,
              "hairline": 0.5,
              "listItemHeight": 108.5,
            },
            "spacing": Object {
              "element": 16,
              "screenEdge": 31.5,
            },
          },
        }
      }
    >
      <Text
        selectable={false}
        style={
          Object {
            "color": "#262835",
            "fontFamily": "System",
            "fontSize": 21.25714285714286,
            "fontWeight": "700",
            "lineHeight": 29.76,
            "textAlign": "left",
          }
        }
      >
        Total
        : 
      </Text>
      <Text
        selectable={false}
        style={
          Object {
            "color": "#262835",
            "fontFamily": "System",
            "fontSize": 21.25714285714286,
            "fontWeight": "400",
            "lineHeight": 29.76,
            "textAlign": "left",
          }
        }
      >
        £0
      </Text>
    </View>
    <View
      flexDirection="row"
      spacing={
        Object {
          "bottom": 3,
        }
      }
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingBottom": 15.942857142857143,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
        ]
      }
      theme={
        Object {
          "colorPalette": Object {
            "green": Array [
              "#0BA866",
            ],
            "grey": Array [
              "#F7F8FC",
              "#EDEDF4",
              "#E1E1EA",
              "#D4D5DD",
              "#BCBCC9",
              "#9699AA",
              "#7C7E8D",
              "#676978",
              "#484A58",
            ],
            "mango": Array [
              "#FFB900",
              "#FFD566",
            ],
            "midnightBlue": Array [
              "#262835",
            ],
            "red": Array [
              "#FFFEF3",
              "#FFE0D7",
              "#FFC2BB",
              "#FFA5A0",
              "#FF8886",
              "#FF6B6C",
              "#FF4D54",
              "#FF333C",
              "#FF1A23 ",
            ],
            "seaGreen": Array [
              "#D5F1F3",
              "#C1EAED",
              "#ACE3E7",
              "#9CDEE1",
              "#82D5DA",
              "#6DCDD3",
              "#59C7CE",
              "#2FB9C2",
              "#238B92",
              "#185D61",
            ],
            "skyBlue": Array [
              "#F7FAFD",
              "#E5F2FC",
              "#C0DEF8",
              "#99CAF4",
              "#75B6EF",
              "#5DA6EC",
              "#4C97E8",
              "#4689DA",
              "#3871C2",
              "#3260B0",
            ],
            "white": Array [
              "#FFFFFF",
            ],
            "yellow": Array [
              "#FFF8E4",
            ],
          },
          "context": Object {
            "colors": Object {
              "actions": Object {
                "background": Object {
                  "link": Object {
                    "complete": "",
                    "default": "transparent",
                    "disabled": "transparent",
                    "error": "",
                    "focused": "transparent",
                    "loading": "transparent",
                  },
                  "primary": Object {
                    "complete": "",
                    "default": "#FF4D54",
                    "disabled": "#E1E1EA",
                    "error": "",
                    "focused": "",
                    "loading": "#E1E1EA",
                  },
                  "secondary": Object {
                    "complete": "",
                    "default": "transparent",
                    "disabled": "transparent",
                    "error": "",
                    "focused": "transparent",
                    "loading": "transparent",
                  },
                },
                "foreground": Object {
                  "link": Object {
                    "complete": "",
                    "default": "#FFFEF3",
                    "disabled": "#262835",
                    "error": "",
                    "focused": "",
                  },
                  "primary": Object {
                    "complete": "",
                    "default": "#FFFFFF",
                    "disabled": "#262835",
                    "error": "",
                    "focused": "",
                  },
                  "secondary": Object {
                    "complete": "",
                    "default": "#FFFEF3",
                    "disabled": "#262835",
                    "error": "",
                    "focused": "",
                  },
                },
              },
              "input": Object {
                "background": Object {
                  "complete": "#F7FAFD",
                  "default": "#F7F8FC",
                  "error": "#FFFEF3",
                  "focused": "#FFFFFF",
                },
                "border": Object {
                  "complete": "#4689DA",
                  "default": "#9699AA",
                  "error": "#FFE0D7",
                  "focused": "#FF4D54",
                },
                "icon": Object {
                  "complete": "#262835",
                  "default": "#262835",
                  "error": "#262835",
                  "focused": "#262835",
                },
                "label": Object {
                  "complete": "#9699AA",
                  "default": "#9699AA",
                  "error": "#FFE0D7",
                  "focused": "#9699AA",
                },
                "outerLabel": Object {
                  "complete": "#262835",
                  "default": "#262835",
                  "error": "#262835",
                  "focused": "#262835",
                },
              },
              "states": Object {
                "error": Object {
                  "complete": "",
                  "default": "#FFFEF3",
                  "error": "",
                  "focused": "",
                },
                "loading": Object {
                  "complete": "",
                  "default": "#FFB900",
                  "error": "",
                  "focused": "",
                },
              },
              "typography": Object {
                "blue": "#3260B0",
                "error": "#FF8886",
                "inverse": "#D4D5DD",
                "link": "#3871C2",
                "mango": "#FFB900",
                "primary": "#262835",
                "secondary": "#F7F8FC",
                "success": "#0BA866",
                "tertiary": "#7C7E8D",
                "white": "#FFFFFF",
              },
            },
            "fontSizes": Object {
              "paragraph": Object {
                "large": 31.5,
                "normal": 27.5,
                "small": 23.5,
              },
              "title": Object {
                "large": 63,
                "normal": 47.5,
                "small": 35.5,
              },
            },
            "fontWeights": Object {
              "paragraph": Object {
                "bold": "700",
                "medium": "600",
                "regular": "400",
              },
              "title": Object {
                "bold": "700",
                "medium": "600",
                "regular": "400",
              },
            },
            "sizes": Object {
              "buttonHeight": 108.5,
              "formElementHeight": 91,
              "formElementIconHeight": 67,
              "hairline": 0.5,
              "listItemHeight": 108.5,
            },
            "spacing": Object {
              "element": 16,
              "screenEdge": 31.5,
            },
          },
        }
      }
    >
      <Text
        selectable={false}
        style={
          Object {
            "color": "#262835",
            "fontFamily": "System",
            "fontSize": 21.25714285714286,
            "fontWeight": "700",
            "lineHeight": 29.76,
            "textAlign": "left",
          }
        }
      >
        Portfolio
        : 
      </Text>
      <Text
        selectable={false}
        style={
          Object {
            "color": "#262835",
            "fontFamily": "System",
            "fontSize": 21.25714285714286,
            "fontWeight": "400",
            "lineHeight": 29.76,
            "textAlign": "left",
          }
        }
      >
        £0
      </Text>
    </View>
  </View>
</View>
`;
